### ISSUE TITLE
node v17: Error loading `tslib` helper library

### ISSUE THREAD

**Issue by aladdin-add** [ezolenko/rollup-plugin-typescript2#286]
#### What happens and why it is wrong

```bash
Error loading `tslib` helper library.
/Users/weiran/repo/dian/tsdx/node_modules/rollup-plugin-typescript2/dist/rollup-plugin-typescript2.cjs.js:25177
    throw e;
    ^

Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './package.json' is not defined by "exports" in /Users/weiran/repo/dian/tsdx/node_modules/rollup-plugin-typescript2/node_modules/tslib/package.json
```

##### Versions

```text
  Node: 17.0.0
  OS: macOS 12.0
  yarn: 3.0.2

   rollup: ^2.57.0 => 2.58.0 
    rollup-plugin-typescript2: ^0.30.0 => 0.30.0 
    typescript: ^4.4.3 => 4.4.4
```

#### rollup.config.js

<details>
  <summary>`rollup.config.js`: </summary>

</details>

#### tsconfig.json

<details>
  <summary>`tsconfig.json`: </summary>

</details>

#### package.json

<details>
  <summary>`package.json`: </summary>

</details>

#### plugin output with verbosity 3

<details>
  <summary>plugin output with verbosity 3: </summary>

</details>

**Comment by aladdin-add** [comment]
this is from https://github.com/weiran-zsd/dts-cli/issues/78.

**Comment by yoeunes** [comment]
Hello @aladdin-add did you find any solution for this error ? I got it also with` node 17`

```
  Node: v17.0.1
  npm: 8.1.0
  OS: Ubuntu
  rollup: 2.58.0
rollup-plugin-typescript2: 0.30.0
```

**Comment by xiCO2k** [comment]
Checking in on this, any updates?

**Comment by bluelovers** [comment]
> Checking in on this, any updates?

i think my pr can fix this

**Comment by bluelovers** [comment]
can try this before merge apply

```
"rollup-plugin-typescript2": "bluelovers/rollup-plugin-typescript2#pr/node-v17-tslib-002",
```

**Comment by ezolenko** [comment]
@bluelovers I'm having trouble building with node 17 (unrelated to your PR), because something breaks semver either at runtime (before upgrading commonjs plugin) or at build time, so I had to disable version check. Do you see those problems at all? 

See this commit, which is basically your PR but with any changes needed to upgrade: https://github.com/ezolenko/rollup-plugin-typescript2/commit/cb2621faccd5e7057a91f575e00171ced3da153c

I used normal resolve (stole it from @rollup/rollup-plugin-typescript):
```
import resolve from "resolve";
const tslibPath = resolve.sync("tslib/tslib.es6.js");
```

Is your approach with `"@yarn-tool/resolve-package"` significantly better?

**Comment by bluelovers** [comment]
u need build at node.js 16 before publish new version once

after this,  ur version test will work

**Comment by aladdin-add** [comment]
a workaround: if you are using docker, you can change the `node:latest` => `node:lts`(or `node:16/14`) to avoid the error.

**Comment by bluelovers** [comment]
> Is your approach with `"@yarn-tool/resolve-package"` significantly better?

@ezolenko u can see source code for watch what i do make it work on node.js 17

 https://github.com/bluelovers/ws-yarn-workspaces/blob/e57f6acafbe7c00c43593823a5d16ada08aeb920/packages/%40yarn-tool/resolve-package/index.ts#L13

**Comment by ezolenko** [comment]
@bluelovers released in 0.31.0, thanks for the PR

**Comment by aladdin-add** [comment]
confirmed it has been fixed in v0.31.0. 🎉

and seems also fixed https://github.com/ezolenko/rollup-plugin-typescript2/issues/267?

**Comment by agilgur5** [comment]
> Is your approach with `@yarn-tool/resolve-package` significantly better?

This seems to have been fixed upstream in `tslib` [`2.4.0`'s `package.json`](https://github.com/microsoft/tslib/blob/a7129c7bd500ce378ec19234247f3d0b1635e89d/package.json#L35) since https://github.com/microsoft/tslib/pull/135 was recently merged and recently. Basically the root cause is that the old way of doing `*` package exports was deprecated in Node 16 (c.f. #267), then removed in Node 17... but the new way didn't work on older Node versions so there were some compatibility issues to work through (now `tslib` uses _both_ the old and new ways).

So if we upgrade `tslib` to `^2.4.0` we should be able to move back to the old method using `require.resolve` and drop the additional dep of `@yarn-tool/resolve-package`.

Also, to workaround prior to `2.4.0` without an additional dep, we could have also resolved to absolute paths, as package exports do not prevent importing internal via absolute paths (they are a ["weak" encapsulation](https://nodejs.org/api/packages.html#main-entry-point-export)).
